tools/serverpod_cli/lib/src/generator/open_api/objects/schema.dart

```diff
@@ -0,0 +1,199 @@
+}
+
+/// Serializes a [TypeDefinition] [List] type to Json
+Map<String, dynamic> listTypeToJson(TypeDefinition type, [bool child = false]) {
+  assert(type.isListType,
+      'Use listTypeToJson only when the typeDefinition is of the ListType.');
+  Map<String, dynamic> map = {};
+  map['type'] = SchemaObjectType.array.name;
+
+  map['items'] = {};
+
+  var generic = type.generics;
+  if (generic.isEmpty) {
+    map['items']['\$ref'] = _getRef('AnyValue');
+    return map;
+  }
+  if (generic.first.isListType) {
+    map['items'] = listTypeToJson(generic.first, true);
+    return map;
+  }
+  if (generic.first.isMapType) {
+    map['items'] = mapTypeToJson(generic.first, true);
+    return map;
+  }
+
+  if (generic.first.isOtherType) {
+    map['items'] = otherTypeToJson(generic.first, true);
+    return map;
+  }
+  map['items'] = dartCoreTypeToJson(generic.first, true);
+  return map;
+}
+
+///  Schema object used within [ComponentObject]
+class ComponentSchemaObject {
+  final SerializableEntityDefinition entityDefinition;
+
+  ComponentSchemaObject(
+    this.entityDefinition,
+  );
+
+  Map<String, dynamic> toJson() {
+    Map<String, dynamic> map = {};
+
+    if (entityDefinition is EnumDefinition) {
+      var enumDefinition = entityDefinition as EnumDefinition;
+      map[entityDefinition.className] = {
+        'type': SchemaObjectType.string.name,
+        'enum': enumDefinition.values.map((e) => e.name).toList(),
+      };
+      return map;
+    }
+    var classDefinition = entityDefinition as ClassDefinition;
+    Map<String, dynamic> objectMap = {};
+    objectMap['type'] = SchemaObjectType.object.name;
+    objectMap['properties'] = {};
+    for (var field in classDefinition.fields) {
+      objectMap['properties'][field.name] = {};
+      if (field.type.isListType) {
+        objectMap['properties'][field.name] = listTypeToJson(field.type, true);
+      } else if (field.type.isMapType) {
+        objectMap['properties'][field.name] = mapTypeToJson(field.type, true);
+      } else if (field.type.isOtherType) {
+        objectMap['properties'][field.name] = otherTypeToJson(field.type, true);
+      } else {
+        objectMap['properties'][field.name] =
+            dartCoreTypeToJson(field.type, true);
+      }
+    }
+    map[entityDefinition.className] = objectMap;
+
+    return map;
+  }
+}
+
+/// A Schema object used within [ContentObject]. Generate request body content
+/// based on the parameters of a Serverpod endpoint's method.
+class RequestContentSchemaObject {
+  final List<ParameterDefinition> params;
+  RequestContentSchemaObject({
+    required this.params,
+  });
+
+  Map<String, dynamic> toJson() {
+    Map<String, dynamic> map = {};
+    map['type'] = SchemaObjectType.object.name;
+    map['properties'] = {};
+    for (var param in params) {
+      map['properties'][param.name] = typeDefinitionToJson(param.type, true);
+    }
+
+    return map;
+  }
+}
+
+/// A Schema Object that will be used in ParameterObject
```

Comment:
Sometimes you have used brackets to refer to other objects, sometimes not. Please look through everything and make sure it is consistent. This should be `[ParameterObject]` and I assume `[SchemaObject]`.